internal/checkers/negative_postive.go

```diff
@@ -0,0 +1,81 @@
+package checkers
+
+import (
+	"golang.org/x/tools/go/analysis"
+
+	"github.com/Antonboom/testifylint/internal/analysisutil"
+)
+
+// NegativePositive detects situations like
+//
+//	assert.Greater(t, 0, value)
+//	assert.Less(t, 0, value)
+//
+// and requires
+//
+//	assert.Positive(t, value)
+//	assert.Negative(t, value)
+type NegativePositive struct{}
+
+// NewNegativePositive constructs NegativePositive checker.
+func NewNegativePositive() NegativePositive { return NegativePositive{} }
+func (NegativePositive) Name() string       { return "negative-positive" }
+
+func (checker NegativePositive) Check(pass *analysis.Pass, call *CallMeta) *analysis.Diagnostic {
+	if len(call.Args) < 2 {
+		// "Greater" and "Less" expect at least 2 arguments
+		// no need to check for assert with less arguments
+		return nil
+	}
+
+	// this variable will help us to consider which argument is zero
+	secondArgZero := isZero(call.Args[1])
+	if !xor(isZero(call.Args[0]), secondArgZero) {
+		// this checker only needed if one of the argument is zero
+		// if both are zeros it will be reported by useless-assert checker
+		return nil
+	}
+
+	// ok now we know one of the argument is zero, but not both
```

Comment:
please, remove useless comments
code must be self-documented and in ideal picture do not contain comments at all

if you see the place for comment use `// NOTE(author): comment` notation

Reply:
That's subjective and open to debate, but I will respect your guidance.